packages/config/federated-modules.js

```diff
@@ -0,0 +1,77 @@
+        ...dependencies.react ? [{ react: { singleton: true, requiredVersion: dependencies.react } }] : [],
+        ...dependencies['react-dom'] ? [{ 'react-dom': { singleton: true, requiredVersion: dependencies['react-dom'] } }] : [],
+        ...dependencies['react-router-dom'] ? [{ 'react-router-dom': { singleton: true, requiredVersion: dependencies['react-router-dom'] } }] : [],
+        ...dependencies['@patternfly/react-table'] ? [{ '@patternfly/react-table': { singleton: true, requiredVersion: dependencies['@patternfly/react-table'] } }] : [],
+        ...dependencies['@patternfly/react-core'] ? [{ '@patternfly/react-core': { singleton: true, requiredVersion: dependencies['@patternfly/react-core'] } }] : [],
+        ...dependencies['@patternfly/react-core'] ? [{ '@patternfly/react-icons': { singleton: true, requiredVersion: dependencies['@patternfly/react-icons'] } }] : [],
+        {
+            '@patternfly/react-tokens': {
+                singleton: true,
+                ...dependencies['@patternfly/react-tokens'] && {
+                    requiredVersion: dependencies['@patternfly/react-tokens']
+                }
+            }
+        },
+        ...dependencies['@redhat-cloud-services/frontend-components'] ? [{
+            '@redhat-cloud-services/frontend-components': {
+                singleton: true,
+                requiredVersion: dependencies['@redhat-cloud-services/frontend-components']
+            }
+        }] : [],
+        ...dependencies['@redhat-cloud-services/frontend-components-utilities'] ? [{
+            '@redhat-cloud-services/frontend-components-utilities': {
+                singleton: true,
+                requiredVersion: dependencies['@redhat-cloud-services/frontend-components-utilities']
+            }
+        }] : [],
+        ...dependencies['@redhat-cloud-services/frontend-components-notifications'] ? [{
+            '@redhat-cloud-services/frontend-components-notifications': {
+                singleton: true,
+                requiredVersion: dependencies['@redhat-cloud-services/frontend-components-notifications']
+            }
+        }] : [],
+        ...dependencies['react-redux'] ? [{ 'react-redux': { singleton: true, requiredVersion: dependencies['react-redux'] } }] : [],
+        ...dependencies['redux-promise-middleware'] ? [{ 'redux-promise-middleware': { singleton: true, requiredVersion: dependencies['redux-promise-middleware'] }
+        }] : []
+    ];
+
+    if (debug) {
+        console.log('Using package at path: ', resolve(root, './package.json'));
+        console.log('Using appName: ', appName);
+        console.log(`Using ${exposes ? 'custom' : 'default'} exposes`);
+        console.log('Number of custom shared modules is: ', shared.length);
+        console.log('Number of default shared modules is: ', sharedDeps.length);
+    }
+
+    return new ModuleFederationPlugin({
+        name: appName,
+        filename: `${appName}.[chunkhash].js`,
```

Comment:
This makes it near impossible for an app to load another app unless the `chunkhash`s for each build are stored in yet another file that is `fetch`ed on each page load.

In managed Kafka's case mk-ui-host would like to load a module from mk-ui-frontend (and it's even kindly abstracted the URL into a `fetch`ed json file). Is there support for this or should our frontends just not use this helper?

Reply:
The reason why we have hashes even in fed modules is that akamai has a nasty habit of serving stale JS assets.

The common config generates a `fedmods.json` file with all necessary metadata to then register the federated module in chrome.

This registration happens only once, on the first app load. We chose to use runtime module registration over build time in order to prevent additional chrome builds whenever any app changes its filenames or new app is added etc.

That said, applications do not have to use the common plugin for federated modules. We realize that many apps may choose different naming schemes. This plugin is optional. You can check the docs here: https://github.com/RedHatInsights/insights-chrome/blob/chrome-2.0/docs/migrationGuide.md#add-federated-modules-plugin